core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -212,6 +212,16 @@ object DataMirror {
     def chiselTypeClone[T <: Data](target: T): T = {
       target.cloneTypeFull
     }
+
+    /** Returns the IOs of a module.
+      *
+      * This method does not necessarily return the final ports of the target module. It consults Chisel's internal data
+      * structures to extract the module's IOs. For this reason, it is generally safer to prefer [[modulePorts]], but
+      * this method may be used for certain use cases that want the current list of ports before the module is closed.
+      *
+      * @param target BaseModule to get IOs from
+      */
+    def moduleIOs(target: BaseModule): Seq[Data] = target.getIOs
```

Comment:
can we call this something like `currentModuleIOs` or `currentModulePorts`? Are we trying to convey a difference between the Ports and IOs returned between the two APIs? Do we not wnat to return the String name yet because that is not finalized...?

Reply:
Yeah I wasn't sure what to call this, I think currentModulePorts is a good name, so we have `DataMirror.modulePorts` and `DataMirror.internal.currentModulePorts`.

I didn't include the string names because I didn't need them, but I can add that for symmetry.

Reply:
Oh, actually, getting the names is a little tricky. It is possible to get the names in atModuleBodyEnd, but only for IOs that were added before the atModuleBodyEnd. I could add the names but users will see exceptions if they call this too early. Since I didn't need them, I am inclined to no add them, and that way this API should always succeed.

Reply:
I will leave the names off for now.